x-pack/plugin/identity-provider/src/test/java/org/elasticsearch/xpack/idp/saml/sp/SamlServiceProviderIndexTests.java

```diff
@@ -160,12 +160,16 @@ public void testWritesViaAliasIfItExists() {
         assertThat(readDocument(document.docId), equalTo(document));
     }
 
-    public void testInstallTemplateAutomaticallyOnClusterChange() {
+    public void testInstallTemplateAutomaticallyOnClusterChange() throws Exception {
         // Create an index that will trigger a cluster state change
-        client().admin().indices().create(new CreateIndexRequest(randomAlphaOfLength(7).toLowerCase(Locale.ROOT))).actionGet();
+        final String indexName = randomAlphaOfLength(7).toLowerCase(Locale.ROOT);
+        client().admin().indices().create(new CreateIndexRequest(indexName)).actionGet();
+
+        ensureGreen(indexName);
 
         IndexTemplateMetaData templateMeta = clusterService.state().metaData().templates().get(SamlServiceProviderIndex.TEMPLATE_NAME);
-        assertNotNull(templateMeta);
+
+        assertBusy(() -> assertThat("template should have been installed", templateMeta, notNullValue()));
```

Comment:
Am I reading this wrong? There is nothing changing in the `assertBusy` here, wouldn't you need to reexecute the template get inside the `assertBusy`?

Reply:
No you're not reading that wrong, I just need more sleep.